### PR TITLE
chore: use `yarn@3.2.3`

### PR DESCRIPTION
**Pull request by magic-akari**
**Description:**

**BREAKING CHANGE:**

**Related issue (if exists):**
- #https://github.com/swc-project/swc/discussions/5614

### CODE REVIEW

**Comment by Brooooooklyn** [comment]
I prefer `nodeLinker: node-modules` config for yarn@3, default `nodeLinker` is pnp

**Comment by Austaras** [comment]
Maybe the simplest solution is revert to yarn 1 and update yarn lock.

**Comment by kdy1** [comment]
Maybe, I changed it to reduce time but if this is problematic I think we can go back to yarn v1

File: .github/workflows/publish-node.yml [hunk 1]
```diff
@@ -228,7 +228,6 @@ jobs:
           image: ${{ matrix.settings.docker }}
           options: -v ${{ env.HOME }}/.cargo/git:/root/.cargo/git -v ${{ env.HOME }}/.cargo/registry:/root/.cargo/registry -v ${{ github.workspace }}:/build -w /build
           run: >-
-            yarn global add wasm-pack &&
```
Comment (magic-akari) [inline .github/workflows/publish-node.yml:231]: https://github.com/swc-project/swc/blob/c746611f262d8b3fbac1c29c23f6c2a4c5f3ce2b/.github/workflows/publish-node.yml#L219-L223

We have installed wasm-pack above these lines.
Is it necessary to install it again in docker step?
@kdy1

Comment (kdy1) [reply .github/workflows/publish-node.yml:231]: I don't think so, seems like a mistake

**Comment by ojkelly** [comment]
> I prefer `nodeLinker: node-modules` config for yarn@3, default `nodeLinker` is pnp

Just for what it's worth, you can use which ever linker you prefer. In most cases `pnp` is good choice, if it's not, or if it introduces some breaking changes you can safely use `node-modules`.

**Comment by magic-akari** [comment]
@Brooooooklyn @ojkelly 

https://github.com/swc-project/swc/blob/360767394fcd99703dc351150d3c9aac66807d38/.yarnrc.yml#L1

It's already `node-modules`

**Comment by magic-akari** [comment]
blocked by nested package structure.
Maybe we should use yarn workspace.

**Pull request closed without merging**